packages/salesforcedx-vscode-apex/src/oas/generationStrategy/methodByMethodStrategy.ts

```diff
@@ -5,23 +5,99 @@
   PromptGenerationResult,
-  PromptGenerationStrategyBid
+  PromptGenerationStrategyBid,
+  OpenAPIDoc
 } from '../../openApiUtilities/schemas';
 import { IMPOSED_FACTOR, PROMPT_TOKEN_MAX_LIMIT, RESPONSE_TOKEN_MAX_LIMIT, SUM_TOKEN_MAX_LIMIT } from '.';
 import { GenerationStrategy } from './generationStrategy';
-
+import { prompts } from './prompts';
 export const METHOD_BY_METHOD_STRATEGY_NAME = 'MethodByMethod';
 export class MethodByMethodStrategy extends GenerationStrategy {
-  callLLMWithPrompts(): Promise<string[]> {
-    throw new Error('Method not implemented.');
+  async callLLMWithPrompts(): Promise<string[]> {
+    try {
+      const llmService = await this.getLLMServiceInterface();
+
+      // Filter valid prompts and map them to promises
+      const responsePromises = this.prompts.filter(p => p?.length > 0).map(prompt => llmService.callLLM(prompt));
+
+      // Execute all LLM calls in parallel and store responses
+      this.llmResponses = await Promise.all(responsePromises);
+
+      return this.llmResponses;
+    } catch (error) {
+      const errorMessage = error instanceof Error ? error.message : String(error);
+      throw new Error(errorMessage);
+    }
   }
-  generateOAS(): Promise<string> {
-    throw new Error('Method not implemented.');
+
+  public async generateOAS(): Promise<string> {
+    const oas = await this.callLLMWithPrompts();
+    if (oas.length > 0) {
+      try {
+        const combinedText = this.combineYamlByMethod(oas);
+        return combinedText;
+      } catch (e) {
+        throw new Error(`Failed to parse OAS: ${e}`);
+      }
+    } else {
+      throw new Error('No OAS generated');
+    }
+  }
+
+  cleanYamlString(input: string): string {
+    return input
+      .replace(/^```yaml\n/, '') // Remove leading triple backtick (if any)
+      .replace(/\n```$/, '') // Remove trailing triple backtick (if any)
+      .replace(/```\n\s*$/, '') // Remove trailing triple backtick with new line (if any)
+      .trim(); // Ensure no extra spaces
+  }
+
+  combineYamlByMethod(docs: string[]) {
+    const combined: OpenAPIDoc = {
+      openapi: '3.0.0',
+      info: {
+        title: this.context.classDetail.name,
+        version: '1.0.0',
+        description: `This is auto-generated OpenAPI v3 spec for ${this.context.classDetail.name}.`
+      },
+      paths: {},
+      components: { schemas: {} }
+    };
+
+    for (const doc of docs) {
+      const yamlCleanDoc = this.cleanYamlString(doc);
+      let parsed = null;
+      try {
+        parsed = yaml.parse(yamlCleanDoc) as OpenAPIDoc;
+      } catch (error) {
+        throw new Error(`Failed to parse YAML: ${error}`);
+      }
+      // Merge paths
+      for (const [path, methods] of Object.entries(parsed.paths)) {
+        if (!combined.paths[path]) {
+          combined.paths[path] = {};
+        }
+        Object.assign(combined.paths[path], methods);
+      }
+      // Merge components
+      if (parsed.components?.schemas) {
```

Comment:
@mingxuanzhangsfdx should this check if the schema already exists, as is being done above with paths